src/openrct2-ui/UiContext.Win32.cpp

```diff
@@ -41,17 +41,26 @@
 
 static std::wstring SHGetPathFromIDListLongPath(LPCITEMIDLIST pidl)
 {
+#    if defined(__MINGW32__)
     std::wstring pszPath(MAX_PATH, 0);
-    while (!SHGetPathFromIDListW(pidl, &pszPath[0]))
+    auto result = SHGetPathFromIDListW(pidl, pszPath.data());
+#    else
+    // Limit path length to 32K
+    std::wstring pszPath(std::numeric_limits<int16_t>().max(), 0);
+    auto result = SHGetPathFromIDListEx(pidl, pszPath.data(), static_cast<DWORD>(pszPath.size()), GPFIDL_DEFAULT);
+#    endif
+    if (result)
     {
-        if (pszPath.size() >= SHRT_MAX)
+        // Truncate at first null terminator
+        auto length = pszPath.find(L'\0');
+        if (length != std::wstring::npos)
         {
-            // Clearly not succeeding at all, bail
-            return std::wstring();
+            pszPath.resize(length);
+            pszPath.shrink_to_fit();
         }
-        pszPath.resize(pszPath.size() * 2);
+        return pszPath;
```

Comment:
We probably want to return an empty one here.

Reply:
Why?

Reply:
Because it did not find a null terminator which means its probably a broken path so handing over so much memory is a waste? I would perhaps even add a Guard::Assert here instead.

Reply:
It wouldn't have returned TRUE if it was a broken path. It just means that the path is exactly MAX_PATH.